tests/e2e/rosa_autoscaler_test.go

```diff
@@ -196,8 +200,435 @@ var _ = Describe("Autoscaler", labels.Feature.Autoscaler, func() {
+						"--scale-down-delay-after-delete", "10s",
+						"--scale-down-delay-after-failure", "10s",
+						"--gpu-limit", "nvidia.com/gpu,0,10",
+						"--gpu-limit", "amd.com/gpu,1,5",
+						"--scale-down-unneeded-time", "10s")
+					Expect(err).To(HaveOccurred())
+					textData := rosaClient.Parser.TextData.Input(resp).Parse().Tip()
+					Expect(textData).To(ContainSubstring("ERR: Hosted Control Plane clusters do not support cluster-autoscaler configuration"))
+
+					By("Describe the autoscaler of the cluster")
+					output, err := rosaClient.AutoScaler.DescribeAutoScaler(clusterID)
+					Expect(err).To(HaveOccurred())
+					textData = rosaClient.Parser.TextData.Input(output).Parse().Tip()
+					Expect(textData).To(ContainSubstring("ERR: Hosted Control Plane clusters do not support cluster-autoscaler configuration"))
+
+					By("Edit the autoscaler of the cluster")
+					resp, err = rosaClient.AutoScaler.EditAutoScaler("-c", clusterID, "--ignore-daemonsets-utilization",
+						"--min-cores", "0",
+						"--max-cores", "10",
+						"--scale-down-delay-after-add", "0s",
+						"--gpu-limit", "amd.com/gpu,1,5")
+					Expect(err).To(HaveOccurred())
+					textData = rosaClient.Parser.TextData.Input(resp).Parse().Tip()
+					Expect(textData).To(ContainSubstring("ERR: Hosted Control Plane clusters do not support cluster-autoscaler configuration"))
+
+					By("Delete the autoscaler of the cluster")
+					resp, err = rosaClient.AutoScaler.DeleteAutoScaler(clusterID)
+					Expect(err).To(HaveOccurred())
+					textData = rosaClient.Parser.TextData.Input(resp).Parse().Tip()
+					Expect(textData).To(ContainSubstring("ERR: Hosted Control Plane clusters do not support cluster-autoscaler configuration"))
+
+				} else if clusterConfig.Autoscaler != nil {
+
+					By("Create the autoscaler to the cluster")
+					resp, err := rosaClient.AutoScaler.CreateAutoScaler("-c", clusterID, "--balance-similar-node-groups",
+						"--skip-nodes-with-local-storage",
+						"--log-verbosity", "4",
+						"--max-pod-grace-period", "0",
+						"--pod-priority-threshold", "1",
+						"--ignore-daemonsets-utilization",
+						"--max-node-provision-time", "10m",
+						"--balancing-ignored-labels", "aaa",
+						"--max-nodes-total", "1000",
+						"--min-cores", "0",
+						"--scale-down-delay-after-add", "10s",
+						"--max-cores", "100",
+						"--min-memory", "0",
+						"--max-memory", "4096",
+						"--scale-down-enabled",
+						"--scale-down-utilization-threshold", "1",
+						"--scale-down-delay-after-delete", "10s",
+						"--scale-down-delay-after-failure", "10s",
+						"--gpu-limit", "nvidia.com/gpu,0,10",
+						"--gpu-limit", "amd.com/gpu,1,5",
+						"--scale-down-unneeded-time", "10s")
+					Expect(err).To(HaveOccurred())
+					textData := rosaClient.Parser.TextData.Input(resp).Parse().Tip()
+					Expect(textData).To(ContainSubstring("ERR: Autoscaler for cluster '%s' already exists", clusterID))
+
+					By("Edit the autoscaler without setting cluster")
```

Comment:
Those below are negative testing. Shouldn't we extract them to a different case ?

Reply:
All are fully negative checkpoints, so it is in one test case